core/storage/config/gae_models.py

```diff
@@ -180,3 +180,68 @@ def create(
             rules=rule_dicts,
             rule_schema_version=rule_schema_version,
             default_value=default_value)
+
+
+class FeatureFlagModel(base_models.BaseModel):
+    """A class that represents named dynamic feature-flag.
+    This model only stores fields that can be updated in run time.
+
+    The id is the name of the feature-flag.
+    """
+
+    force_enable_for_all_users = datastore_services.BooleanProperty(
+        default=False, indexed=True)
+    rollout_percentage = datastore_services.IntegerProperty(
```

Comment:
It's often conventional to use a range from 0 to 1000 for rollout percentage (so that we can have granularity like 12.5%). While we probably only will use increments of 10 for now, having the flexibility for sub-percentage rollouts will be useful if our userbase grows large enough where sub-percentage slices of the audience are statistically significant.

Should this also be documented to explain its range?

Reply:
@lkbhitesh07 Just a note, I followed up with @BenHenning on this and you don't need to do the first paragraph (but please do the second). The rationale is that it's hard to find a good name for the "1000" case, and the extra granularity probably won't be needed for a while anyway, and the migration path is straightforward if we ever need to do it in future (i.e. just go to the feature flag page and multiply all values by 10 manually, and save the changes).

So, please skip the first paragraph, but please do the second (i.e. documenting the range). Thanks!

Reply:
Thanks @seanlip @BenHenning.
In the UI we do mention the range of the rollout-percentage, just after the "Rollout percentage for logged-in users". Do we plan to mention it somewhere else as well?

Reply:
Yup, as a comment above this line.